core/consensus/beefy/impl/beefy_impl.cpp

```diff
@@ -393,9 +407,39 @@ namespace kagome::network {
     return outcome::success();
   }
 
+  outcome::result<void> BeefyImpl::fetchHeaders() {
+    if (not fetching_headers_ or not beefy_genesis_) {
+      return outcome::success();
+    }
+
+    while (fetching_headers_) {
+      auto& fetchingHeader = *fetching_headers_;
+      if (fetchingHeader <= *beefy_genesis_) {
+        fetching_headers_.reset();
+        break;
```

Comment:
Don't need `fetchHeadersBack` in this case
```suggestion
        return;
```